### PR TITLE
Backported recent metadata pr from 2.27 to 2.26

### PR DESCRIPTION
**Pull request by sultanahamer**
Backported recent metadata pr from 2.27 to 2.26

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/sync/MetadataSyncImportHandler.java [hunk 1]
```diff
@@ -99,7 +100,8 @@ public MetadataSyncSummary importMetadata( MetadataSyncParams syncParams, String
         catch ( Exception e )
         {
             String message = "Exception occurred while trying to import the metadata. " + e.getMessage();
-            log.error( message, e );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/sync/MetadataSyncImportHandler.java]: Use "ex" as variable name

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/webmessage/utils/WebMessageParseUtils.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        {
+            return JSON_MAPPER.readValue( responseNode.toString(), klass );
+        }
+        catch ( IOException e )
+        {
+            throw new WebMessageParseException( "Could not parse the JSON." + e.getMessage(), e );
+        }
+    }
+
+
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/webmessage/utils/WebMessageParseUtils.java]: Extra space, remove

File: dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CrudControllerAdvice.java [hunk 3]
```diff
@@ -227,6 +230,27 @@ public void handleDhisVersionMismatchException( DhisVersionMismatchException ver
         webMessageService.send( WebMessageUtils.forbidden( versionMismatchException.getMessage() ), response, request );
     }
 
+    @ExceptionHandler( MetadataImportConflictException.class )
+    public void handleMetadataImportConflictException( MetadataImportConflictException conflictException, HttpServletResponse response, HttpServletRequest request )
+    {
+
+        if ( conflictException.getMetadataSyncSummary() == null )
```
Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CrudControllerAdvice.java]: Codestyle, always use brackets { } even for one line if statements

File: dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/metadata/sync/MetadataSyncController.java [hunk 4]
```diff
@@ -94,22 +110,36 @@ public MetadataSyncSummary metadataSync() throws MetadataSyncException, DhisVers
...
         }
 
-        return metadataSyncSummary;
+        return new ResponseEntity<MetadataSyncSummary>( metadataSyncSummary, HttpStatus.OK );
+    }
+
+    private void validateSyncSummaryResponse( MetadataSyncSummary metadataSyncSummary ) throws MetadataImportConflictException
+    {
+        ImportReport importReport = metadataSyncSummary.getImportReport();
+        if(importReport.getStatus() != Status.OK){
```
Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/metadata/sync/MetadataSyncController.java]: Codestyle, bracket on new line

**Comment by sultanahamer** [comment]
Made the above mentioned changes @larshelge .

**Comment by larshelge** [comment]
Thanks! Ready for merge.

**Pull request merged**